src/storage/engine/mod.rs

```diff
@@ -71,6 +72,11 @@ pub enum Modify {
 
 pub trait Engine: Send + Debug {
     fn async_write(&self, ctx: &Context, batch: Vec<Modify>, callback: Callback<()>) -> Result<()>;
+
+    fn async_ingest(&self, _: &Context, _: SSTMeta, _: Callback<()>) -> Result<()> {
```

Comment:
I don't think it is proper to define the `async_ingest` in the `Engine`.
I suggest constructing a `RaftCmd` in the `ImportSSTService` and send it to the raftstore.

Reply:
Will try.